### PR TITLE
AgentComponent to forward all state notifications

### LINKED ISSUE #3087: agent side state notifications are not forwarded to client side

### PR DESCRIPTION
**Pull request by andre-merzky**
Fix forwarding of agent side state notifications.  This fixes #3087

### CODE REVIEW

File: src/radical/pilot/pmgr/launching/base.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def stop(self):
 
 # ------------------------------------------------------------------------------
 #
-class PMGRLaunchingComponent(rpu.Component):
+class PMGRLaunchingComponent(rpu.AgentComponent):
```
Comment (mtitov) [inline src/radical/pilot/pmgr/launching/base.py:109]: `ClientComponent`?

**Review by mtitov — COMMENTED** [review]
added one comment and have couple questions regarding `rpu.Worker`
- do we keep it for `Agent_0` and `Agent_N`?
- for `rp.worker.Stager`, should we switch to `ClientComponent`?

File: src/radical/pilot/pmgr/launching/base.py (hunk 1 above)
Comment (andre-merzky) [reply src/radical/pilot/pmgr/launching/base.py:109]: Oh, definitely!  Thanks

**Comment by andre-merzky** [comment]
> added one comment and have couple questions regarding `rpu.Worker`
> 
>     * do we keep it for `Agent_0` and `Agent_N`?
> 
>     * for `rp.worker.Stager`, should we switch to `ClientComponent`?

Yeah, we can probably get rid of the Worker class by now...

**Comment by mtitov** [comment]
as an intermediate testing - RP from this branch runs successfully on Lassen (`tmgr.wait_tasks(uids=tasks_uids, state=rp.AGENT_EXECUTING)` waits correctly)

File: src/radical/pilot/pmgr/launching/base.py (hunk 1 above)
Comment (andre-merzky) [reply src/radical/pilot/pmgr/launching/base.py]: Fixed

**Review by mtitov — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**